README.md

```diff
@@ -1,39 +1,89 @@
-# Role Name
-
-A brief description of the role goes here.
+# zfs-snap-manager
 
+This role installs and configures [zfs-snap-manager](https://github.com/khenderick/zfs-snap-manager), a tool to automatically create, replicate and destroy ZFS snapshots.
 
 ## Requirements
 
-Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance, if the role uses the EC2 module, it may be a good idea to mention in this section that the boto package is required.
-
+`python2` has to be installed on the target machine. Obviously, any ZFS filesystems that should be snapshotted must exist.
+The machine that executes the playbook needs to have internet access as well as `git` installed.
 
 ## Role Variables
 
-A description of the settable variables for this role should go here, including any variables that are in defaults/main.yml, vars/main.yml, and any variables that can/should be set via parameters to the role. Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
+| Name                         | Mandatory / Default      | Description                                                                             |
+| ---------------------------- | ------------------------ | ------------------------------------------------------------------------------- |
+| `global_cache_dir`           | $ansible_env.HOME/.cache | cache directory to clone git repo to |
+| `zfs_snap_manager_version`   | `0.1.13`                 | [zfs-snap-manager](https://github.com/khenderick/zfs-snap-manager) version to use                                                                                         |
+| `zfs_parent_fs`              |                          | Parent pool or filesystem (optionally prepended to all datasets)                     |
+| `zfs_snapshot_defaults`      | `{}`                     | Default snapshot configuration used when there is no explicit configuration for a filesystem, see [snapshot configuration](#snapshot configuration) for content.          |
+| `zfs_filesystems`            | `[]`                     | List of filesystems to be snapshotted. Snapshots can be configured using the dict item `snapshots`. For its content see [snapshot configuration](#snapshot configuration) |
+| `zvols`                      | `[]`                     | List of zvols to be snapshotted. Snapshots can be configured using the dict item `snapshots`. For its content see [snapshot configuration](#snapshot configuration)                |
```

Comment:
Is `zvols` a common abbreviation or parameter outside of the ZFS context? I think it is unique enough.